compatibility.bs

```diff
@@ -832,6 +846,142 @@ supported on all {{Window}} objects and <code><{body}></code> elements as attrib
   </tbody>
 </table>
 
+<h2 id="sa-heuristics-section">Storage Access Heuristics</h2>
+
+Browsers are working to deprecate third-party cookies while avoiding user-facing
+breakage by removing support for common web patterns, particularly login flows via
+third-party identity providers. Browsers can observe <code>Storage access heuristics</code>
+from the user in order to provide short-term storage access to unbreak these flows.
+
+The following spec represents the Chromium implementation of this feature.
+
+<h3 id="sa-heuristics-constants">Constants</h3>
+
+The <dfn>popup heuristic grant duration</dfn> is an [=implementation-defined=]
+[=duration=] that represents the length of time after the popup heuristic is
+detected that the popup [=host=] will have access to the opener [=host=].
+
+Note: 30 days is a reasonable [=popup heuristic grant duration=] value.
+
+The <dfn>redirect heuristic grant duration</dfn> is an [=implementation-defined=]
+[=duration=] that represents the length of time after the redirect heuristic is
+detected that the bounced [=host=] will have access to the final [=host=].
+
+Note: 15 minutes is a reasonable [=redirect heuristic grant duration=] value.
+
+<h3 id="sa-heuristics-grant">Storage Access Grant</h3>
+
+Define a [=powerful feature=] identified by the [=powerful feature/name=]
+"storage-access-heuristics", with the following <dfn>permission key type</dfn>:
+
+A [=permission key=] of the "storage-access-heuristics" feature is a [=tuple=] consisting
+of a [=host=] <dfn property>top-level</dfn> and a [=host=] <dfn property>requester</dfn>.
+
+<div algorithm>
+
+To <dfn>grant access for heuristics</dfn> given a [=host=] |host|, [=host=]
+|firstPartyHost|, and [=duration=] |duration|, perform the following steps:
+
+1. Let |key| be a [=permission key=] with <var ignore>top-level</var> set to |firstPartyHost|
+    and <var ignore>requester</var> set to |host|.
+1. Queue a task on the [=current settings object=]'s [=responsible event loop=] to:
```

Comment:
Queue a task should link to HTML's "queue a global task". See how other specs use it: https://dontcallmedom.github.io/webdex/q.html#queue%20a%20global%20task%40%40html%25%25dfn.

You'll also need to queue to a proper event loop. For example, "Grant access [...]" is called:
 - In a context where it doesn't look like there is an event loop running(?)
    - At least HTML doesn't seem to rely on it; it grabs the "relevant global object" directly from a given document (https://html.spec.whatwg.org/multipage/interaction.html#activation-notification), so we should follow suit
 - ... for a specific document

I think I'd recommend passing in the specific document here, and queueing the global task to that document's window's event loop.

Reply:
Ack. Sorry, I ended up removing the permissions API algorithm entirely in favor of the new global map. Still, I appreciate the helpful context for future specs!